### PR TITLE
Keep the cart after placing an order from the product page

### PR DESCRIPTION
**Pull request by johnmccombs1**
Asana task: https://app.asana.com/0/564264490825835/1117712752858652

This PR adds custom order confirmation page

### CODE REVIEW

File: app/code/community/Bolt/Boltpay/etc/config.xml [hunk 1]
```diff
@@ -176,13 +176,6 @@
 
   <admin>
     <routers>
```
Comment (daisy1754) [inline app/code/community/Bolt/Boltpay/etc/config.xml]: can you send me separate PR for this? assuming it's not related to product page checkout

File: app/code/community/Bolt/Boltpay/Helper/GeneralTrait.php [hunk 2]
```diff
@@ -150,6 +150,16 @@ public function isShoppingCartPage()
...
     }
 
+    /**
+     * @return bool
+     */
+    public function isProductPageCheckout()
+    {
+        $checkoutSession = Mage::getSingleton('checkout/session');
+
+        return $checkoutSession->getData('checkout_type') == Bolt_Boltpay_Block_Checkout_Boltpay::CHECKOUT_TYPE_PRODUCT_PAGE ? true : false;
```
Comment (matt-thomason) [inline app/code/community/Bolt/Boltpay/Helper/GeneralTrait.php]: This could be simplified as `return $checkoutSession->getData('checkout_type') == Bolt_Boltpay_Block_Checkout_Boltpay::CHECKOUT_TYPE_PRODUCT_PAGE;`

File: app/code/community/Bolt/Boltpay/etc/config.xml (hunk 1 above)
Comment (johnmccombs1) [reply app/code/community/Bolt/Boltpay/etc/config.xml]: > LGTM, can you attach screenshot of success screen?

See the screenshot here https://screencast.com/t/yQvvWj7es

File: app/code/community/Bolt/Boltpay/Helper/GeneralTrait.php (hunk 2 above)
Comment (johnmccombs1) [reply app/code/community/Bolt/Boltpay/Helper/GeneralTrait.php]: Thank @matt-thomason, I've updated it.

File: app/code/community/Bolt/Boltpay/controllers/OnepageController.php [hunk 3]
```diff
@@ -92,4 +92,29 @@ public function saveShippingMethodAction()
             throw $e;
         }
     }
+
+    /**
+     * Order success action
+     */
+    public function successAction()
```
Comment (mccottry) [inline app/code/community/Bolt/Boltpay/controllers/OnepageController.php]: Unfortunately, we are overriding the same method here and there is no clear way to distinguish if this is a product page order or a normal order.  

https://github.com/BoltApp/bolt-magento1/blob/cadf0069d45a6570253310aa39a9c40a3f717cf9/app/code/community/Bolt/Boltpay/controllers/OnepageController.php#L43-L80

Since you wish to pursue this without adding something to the DB then my suggestion is to pass a URL GET paramater for `is_bolt_pdp`.  However, by far, I think the far better idea is to extend the order model table with this `is_bolt_pdp`.  With this flag, you will not need to override the success URL at all and we will always be able to clearly tell which orders came from product page and which from some other source.  Is there a reason why you are not wanting to extend the order model?
